### ISSUE TITLE
Defer Solutions at the end of the exercise sheet

### ISSUE THREAD

**Issue by tristelune1** [Strauman/exercisebank#106]
Hi, 

sometimes I need to put the solution at the end of the document. Until now I used a Makefile, but it doesn't work very well. Here is what I did:

- `make sheet`: create the exercise sheet without the solutions
- `make solutions`: create a document with only the solutions. Basically it uses your command `\SolutionsOnly`

My solution is also based on the package etoolbox. I use if statements depending on the argument of the Makefile. 

After I merge the two documents. The problem: the page numbering is not right. The page numbering begins with 1 on the document with solutions, it should be one more than the last page of the exercise sheet. 

Perhaps it can be solved manually, but in order to work for every documents, a command like `\SolutionsEnd` would be interesting to put all solutions at the end of the document. 

There is something similar in the package `exframe`. You can have a look at the documentation on page 4:

[exframe.pdf](http://www.pirbot.com/mirrors/ctan/macros/latex/contrib/exframe/exframe.pdf)

Perhaps it could be interesting to implement something like 

    solution=position

in `exercisebanksetup`,  where we could have the possibilities for `position`

- none : without the solutions
- only: only the solutions
- end: all solutions at the end of the document

Or something you can think that make more sense.

**Comment by Strauman** [comment]
I'm confused? The problem is that the pagenumbering is wrong? How do I reproduce this?

**Comment by tristelune1** [comment]
You can't with only one compilation. 

My goal: to have all exercises at the begin of the exercise sheet and all solutions at the end. I don't know how I can get such a document. So I produce two documents:

- first document: all chosen exercises without the solutions. I save the document, let's say exercise_sheet.pdf The numbering begins with 1.
- second document: all chosen exercises with the option \SolutionOnly. Of course the page numbering begin with 1. I save under the name exercise_sheet_solutions.pdf.

Then with a utility like `pdfunite` I merge the two documents. Of course the page numbering is wrong. 

Perhaps other similar problems could arise, but it's my only example at the moment. 

There is for sure a fix: there is probably a counter somewhere where I could set the page numbering when I compile the document with only the solutions. But:

- I have to do it manually
- it depends on the document

So I can't automate it for every documents. 

Just tell me if something is not clear.

**Comment by Strauman** [comment]
Ah. It's kinda related to #3, which I decided not to do.
Because of the `comments` package, one cannot send arguments to the `\begin{problem}` environment. The alternative would be to do something with `\nextproblem{solution=SOLNAME}` and then have a `\nextsolution{name=SOLNAME}`. However, it would be very much work and would be incredibly slow. I don't really know how to fix this. But I'll keep it in the back of my mind.

**Comment by Strauman** [comment]
Did you maybe consider using some perl-script to merge the LaTeX-files perhaps?

**Comment by tristelune1** [comment]
Ok, I see the problem. So the best is: keep it in your mind and if you get a solution in the future, you could implement it. 

It's not a problem to merge the two files. On linux I can use pdfunite. The problem is how I can automate it. I will find a solution and perhaps it could be added in the documentation for the moment. It could interest other users. As soon as I have something, I'll tell you.

**Comment by Strauman** [comment]
Wait. No. You mean just _printing_ the solutions at the end? That might actually be possible...

**Comment by tristelune1** [comment]
Take a look at the document I have sent to you. I want the same with the correct page numbering. I don't know exactly what you mean by _printing the solutions_ .

**Comment by Strauman** [comment]
Wouldn't the following work?
```
\begin{document}
    \buildset{setName}
    \SolutionsOnly
    \buildset{setName}
\end{document}
```

**Comment by tristelune1** [comment]
Yes, after a first look it seems alright! I test is more extensively tomorrow I confirm you. By the way it seems there is still a problem with the spacing. But that's for another issue.
